crates/agentgateway/src/http/csrf.rs

```diff
@@ -0,0 +1,158 @@
+use ::http::{Method, StatusCode, header};
+
+use crate::http::{PolicyResponse, Request, filters};
+use crate::*;
+
+#[apply(schema)]
+pub struct Csrf {
+	#[serde(default)]
+	additional_origins: Vec<String>,
```

Comment:
```suggestion
	additional_origins: HashSet<String>,
```

Then we can do O(1) lookup instead of O(n)